src/Entity/AbstractRectorRun.php

```diff
@@ -27,9 +29,33 @@ public function getUuid(): Uuid
         return $this->uuid;
     }
 
+    /**
+     * @return DiffSnippet[]
+     */
+    public function getDiffSnippets(): array
+    {
+        $diffSnippets = [];
+
+        $contentDiffsRaw = Strings::split($this->getContentDiff(), '#@@ #');
+        foreach (array_filter($contentDiffsRaw) as $contentDiffRaw) {
+            [$lines, $content] = Strings::split($contentDiffRaw, "# @@\n#");
```

Comment:
it seems cause error:

```
 Undefined array key 1 
```

![Screenshot 2024-08-02 at 18 45 25](https://github.com/user-attachments/assets/f0575b7a-2f20-40a5-b12c-c80074e9d45b)

with code example:

```php
<?php

namespace Miofair\LayoutBundle\Tests\Form\Type;

use PHPUnit\Framework\MockObject\MockObject;
use PHPUnit\Framework\TestCase;

class OnCallbackTest extends TestCase
{
    private (\Exception&MockObject)|null $mock = null;

    protected function setUp(): void
    {
        $this->mock = $this->createMock(\Exception::class);
    }

    public function testConfigureOptions(): void
    {
        $matcher = static::exactly(6);
        $this->mock
            ->expects($matcher)
            ->method('__toString')
            ->willReturnCallback(function (){
                return $this->mock;
            })
        ;
    }
}
```

and code config:

```php
<?php

use Rector\Config\RectorConfig;
use Rector\TypeDeclaration\Rector\Property\TypedPropertyFromAssignsRector;

return RectorConfig::configure()
    // A. whole set
 //   ->withPreparedSets(typeDeclarations: true)
    // B. or few rules
    ->withRules([
        TypedPropertyFromAssignsRector::class
    ]);
```

Reply:
It happen when content result is:

```
"// no change"
```

Reply:
https://github.com/rectorphp/getrector-com/pull/2488